cs3api4lab/api/cs3_ocm_share_api.py

```diff
@@ -147,21 +144,28 @@ def update_received(self, ocm_share_id, field, value):
             self._handle_error(response, "Error updating OCM received share:")
         return response.opaque
 
-    def list(self, share_id=None):
-        if share_id is None:
-            return self._list_ocm_shares()
-        else:
-            return self._get_ocm_share(share_id)
+    def list(self, optional_filter={}):
```

Comment:
Then in this method, the optional filter parameter is passed directly to `ListOCMSharesRequest`.

Otherwise, you are building a method that accepts some kind of filtering, which is great, but in reality you only accept a resource filter (which needs to be converted).

So, this method is generic, and you create a separate method (the `self.share_api.file_shares` I mentioned above) that calls this one and sets the filter appropriate to the use case.